### ISSUE TITLE
Default language

### ISSUE THREAD

**Issue by DavidMStraub** [gramps-project/web-api#85]
Although the new metadata endpoint returns the default locale fixing #73, this is unfortunately not enough for an application to fetch the strings in the default language. As an example, my locale is `de_DE`, and this is what I get for `locale.lang` under `/metdata`, but the corresponding Gramps language (and the language accepted by the `translations` endpoint) is `de`. In principle the application could also fetch the list of translations from `/translations` and do some internal logic, but that would require 3 HTTP calls just to get the strings, and would duplicate code that is in Gramps anyway. I also wouldn't like to simply take the first to characters of the locale, since Gramps *does* distinguish between `pt_BR` and `pt_PT`, and some other examples.

So in my opinion we should add a second key to the `locale` dict in the metadata endpoint given the actual Gramps language the default locale implies.

I think it would be essentially obtained with this code from `grampslocale.py`:

```python
        try:
            lang = _LOCALE_NAMES[lang_code][2]
        except KeyError:
            try:
                lang = _LOCALE_NAMES[lang_code[:2]][2]
            except KeyError:
                LOG.debug("Gramps has no translation for %s", lang_code)
                lang = None
        except IndexError as err:
            LOG.debug("Bad Index for tuple %s\n" % _LOCALE_NAMES[lang_code][0])
            lang = None
```

where `lang_code` is what is currently returned as `language`.

**Comment by DavidMStraub** [comment]
On a related note, why are the underscores in locales replaced with hyphens in the current implementation of `/api/translations/`?

**Comment by cdhorn** [comment]
Yeah I will undo that.  Are the locales not iso codes and the iso codes actually use - in them?  Or am I mixing up two or three different things? I think when I did that I set it to accept either, but yeah it should be consistent.

**Comment by cdhorn** [comment]
So as best I could tell this indicates language codes use a dash: 'https://en.wikipedia.org/wiki/IETF_language_tag'

But this indicates POSIX platforms don't follow that standard: 'https://en.wikipedia.org/wiki/Locale_(computer_software)#POSIX_platforms'

In the Python locale library the getdefaultlocale method in here: 'https://docs.python.org/3/library/locale.html'

Actually references the original RFC which uses a dash here: 'https://tools.ietf.org/html/rfc1766.html'

Are they interchangeable to some extent or no?

**Comment by DavidMStraub** [comment]
Wow, I was not aware of this bewildering variety of standards. But now that we have the default *language* in the same format that Gramps uses, I'd say the *locale* is mostly useful as information about the server configuration, so for me it would be fine just spitting out the locale string as it is, what do you think?

**Comment by Nick-Hall** [comment]
> I also wouldn't like to simply take the first to characters of the locale, since Gramps does distinguish between pt_BR and pt_PT, and some other examples.

We also have a new Austrian translation in PR [#1145](https://github.com/gramps-project/gramps/pull/1145).  So I will probably be creating a `de_AT`.  If I rename the existing `de` translation to `de_DE`, I need to check which will be used if a user just specifies `de`.

**Comment by DavidMStraub** [comment]
> We also have a new Austrian translation in PR [#1145](https://github.com/gramps-project/gramps/pull/1145). So I will probably be creating a `de_AT`. If I rename the existing `de` translation to `de_DE`, I need to check which will be used if a user just specifies `de`.

Interesting! Although I'll definitely not be a user of this :wink:, it's great that there are so many options for localization.

To keep backwards compatibility, I think it would be nice to have checks like for English currently in `grampslocale.py`:

```python
            if locale[:2] == ('en') and locale[:5] != 'en_US':
                return ('en_GB', '1252')
# ...
    if locale.startswith('English') and locale != 'English_United States':
        return ('en_GB', '1252')
# ...
            if not lang and locale[0].startswith('en'):
                locale = ('en_GB', 'UTF-8')
                lang = 'en_GB'
# ...
        if locale[:2] == 'en' and locale[:5] != 'en_US':
            return 'en_GB'

```

where obviously `de_DE` should be the default for `de` (no offense to our Austrian friend :smile:)

**Comment by romjerome** [comment]
I remember an old issue on the localized date handler for french locales (countries : _FR, BE, CA, LU, CH, SN, VU, MC, etc._). Switzerland "needs" something like a german date format (with a dot after the day)... So, the localized date handler/displayer for french locales also supports a typical date format for countries writting/speaking german (just for **fr_CH**, not **fr_LU**)! I suppose the default date formats in UK and in USA could be a more accessible/global sample. Anyway the issue was on LC_ handling as we have to install support for these countries during tests, and French locale could be used by many countries...

**Comment by cdhorn** [comment]
Closing as fixed in #86 and tweaked in #87
